.gitignore

```diff
@@ -5,3 +5,6 @@ coverage
 
 # In case contributors use yarn (just prevent it from being committed). Standardizing on npm for broad compatibility.
 yarn.lock
+
+# Ignore autogenerated .d.ts (and associated .map) files since this are automatically generated prior to publish to NPM.
```

Comment:
You shouldn’t ignore this file. Most repos commit this file

Reply:
Would the expectation w/ [vite-plugin-dts](https://www.npmjs.com/package/vite-plugin-dts) be to run that during development and to also commit changes to those `.d.ts` files? Right now I'm still not convinced we should commit those. 

Again, the reason why is because I do not want to incorporate that into every PR, every commit and etc. The dev workflow is far easier right now keeping it with JSDoc and only instead pushing out the `.d.ts` on publish for developer convenience (in case they are in your situation where JSDoc for some reason doesn't work in your IDE). 

For example, in Jetbrains (PhpStorm is what I'm using), the JSDoc types are pulled in just fine. It's possible this is an issue with VSCode or maybe more specifically your extensions in VSCode, I'm not sure. 

Finally:

> It's easy to directly link issues in typescript hint files, instead of going through the codebase to find out where it was declared.

This may just be a difference of opinion. This is actually what happens for me _**if**_ the `.d.ts` are present; it requires hunting because you get linked to the `.d.ts` file instead of the `.js` file. This is possibly an argument to _not_ include them in the published packages, since 99% of folks using it will not be using a cloned/linked version but instead one with the combined `.js` paired with the `.d.js` files.